services/department.js

```diff
@@ -0,0 +1,79 @@
+const DepartmentSchema = require("../models/department");
+const EmployeeSchema = require("../models/employee");
+const FileService = require("./file");
+const CustomHTTPError = require("../errors");
+
+const mongoose = require("mongoose");
+
+class DepartmentService {
+  async createNewDepartment(department, files) {
+    const { name } = department;
+    const departmentDuplicate = await DepartmentSchema.findOne({ name });
+    if (departmentDuplicate) {
+      throw CustomHTTPError.BadRequest(`Department with the specified name (${name}) already exists.`);
+    }
+    let createdDepartment;
+    if (files) {
+      const fileName = FileService.saveFile(files.picture);
+      createdDepartment = await DepartmentSchema.create({ ...department, picture: fileName });
+    } else {
+      createdDepartment = await DepartmentSchema.create({ ...department });
+    }
+    return createdDepartment;
+  }
+  async getDepartments(query) {
+    const page = parseInt(query.page) || 1;
+    const limit = parseInt(query.limit) || 10;
+    const sortBy = ["asc", "desc"].indexOf(query.sort) < 0 ? "asc" : query.sort;
+    const search = query.q;
+    const filter_query = search ? { name: new RegExp("^" + search, "i") } : {};
+    const total = await DepartmentSchema.find({}).where(filter_query).count();
+    const results = {
```

Comment:
since names and variables have the same name

const results = {
      currentPage: page,
      limit,
      sortBy,
      search,
      total,
    };